Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -123,9 +123,24 @@ public bool IsGlobal
         /// </summary>
         public int NumOfDimensions
         {
-            get { return Dimensions; }
+            get
+            {
+                if (!isInitialized)
+                {
+                    Debug.LogError($"Cannot access Interactable.NumOfDimensions because {gameObject.name}  has not initialized. Ensure Initialize() has been called");
+                    return -1;
+                }
+
+                return Dimensions;
+            }
             set
             {
+                if (!isInitialized)
+                {
+                    Debug.LogError($"Cannot access Interactable.NumOfDimensions because {gameObject.name}  has not initialized. Ensure Initialize() has been called");
```

Comment:
Would it make sense to automatically call `Initialize()` here (and in the other cases that added an error log)?
Then, `Initialize()` would be `private`/`protected` and called either in `Awake()` or when one of these properties is accessed, whichever comes first.

Reply:
Agree - we can do a EnsureInitialized() in case those get / sets are called :)